cachito/workers/tasks/general.py

```diff
@@ -18,10 +18,48 @@
     "failed_request_callback",
     "fetch_app_source",
     "set_request_state",
+    "get_request_state",
+    "check_state",
 ]
 log = logging.getLogger(__name__)
 
 
+def get_request_state(request_id):
+    """
+    Get the state of the request.
+
+    :param int request_id: the Cachito request ID this is for
+    """
+    from cachito.workers.requests import requests_auth_session
+
+    config = get_worker_config()
+    request_url = f'{config.cachito_api_url.rstrip("/")}/requests/{request_id}'
+
+    rv = requests_auth_session.get(request_url)
```

Comment:
Is it expected to separate tasks from the Cachito internal and interact by the Web API?

I'm thinking the state change could reuse existing code as much as possible without the logical separation. For example:

```python
if Request.query.get_or_404(request_id).is_in_progress:
    return
```

On the other hand, if the separation is required, it would be good to have a Cachito lib wrapping the API, then we can:

```python
request = cachito.get_request(request_id)
if request.is_in_progress:
    return
```

Reply:
Tasks are executed on workers, workers do not have direct access to the database. That's why it's necessary to query the Cachito API.

Reply:
I would leave the API wrapper for a separate story, @tkdchen you can file something if you want